src/internal/connector/onedrive/permission.go

```diff
@@ -167,30 +177,54 @@ func diffPermissions(
 	return added, removed
 }
 
-// RestorePermissions takes in the permissions that were added and the
-// removed(ones present in parent but not in child) and adds/removes
-// the necessary permissions on onedrive objects.
-func RestorePermissions(
+func computeParentPermissions(itemPath path.Path, folderMetas map[string]Metadata) (Metadata, error) {
+	var (
+		parent path.Path
+		meta   Metadata
+
+		err error
+		ok  bool
+	)
+
+	parent = itemPath
+
+	for {
+		parent, err = parent.Dir()
+		if err != nil {
+			return Metadata{}, clues.New("getting parent")
+		}
+
+		onedrivePath, err := path.ToOneDrivePath(parent)
+		if err != nil {
+			return Metadata{}, clues.New("get parent path")
+		}
+
+		if len(onedrivePath.Folders) == 0 {
+			return Metadata{}, nil
+		}
+
+		meta, ok = folderMetas[parent.String()]
+		if !ok {
+			return Metadata{}, clues.New("no parent meta")
+		}
+
+		if meta.SharingMode == SharingModeCustom {
+			return meta, nil
+		}
+	}
+}
+
+func updatePermissions(
 	ctx context.Context,
 	creds account.M365Config,
 	service graph.Servicer,
 	driveID string,
 	itemID string,
-	meta Metadata,
+	permAdded, permRemoved []UserPermission,
+	usePermissionIDMappings bool,
```

Comment:
add a comment to the function for why we take in a bool flag (appears it's so we can reuse this code for tests?)

Reply:
I've removed it altogether, it now just computes the mapping from the set to be removed and passes that into the function.